charts/internal/shoot-cert-management-seed/templates/deployment.yaml

```diff
@@ -82,6 +82,7 @@ spec:
         - --allow-target-issuers
         {{- end }}
         - --lease-name=shoot-cert-service
+        - --lease-resource-lock={{ .Values.leaderElection.resourceLock }}
```

Comment:
IIUC this PR couples the leader election resource lock of the extension controller itself with the resource lock of a component that the extension deploys. If this is the case, why it is done like this? Why the extension does not hardcode the resource lock of the component that it deploys (as for example the provider extensions do)?

Reply:
Ok, if this is good practice I can simplify the code.